transport/httptransport/libp2p_server.go

```diff
@@ -183,12 +174,43 @@ func (s *Libp2pCarServer) checkAuth(r *http.Request) (string, *AuthValue, *httpE
 	return authToken, val, nil
 }
 
-func (s *Libp2pCarServer) serveContent(w http.ResponseWriter, r *http.Request, authToken string, val *AuthValue, bic *car.BlockInfoCache) error {
+func (s *Libp2pCarServer) serveContent(w http.ResponseWriter, r *http.Request, authToken string, val *AuthValue) error {
 	ctx := r.Context()
 
-	// Create a CarOffsetWriter and a reader for it
-	cow := car.NewCarOffsetWriter(val.PayloadCid, s.bstore, bic)
-	content := car.NewCarReaderSeeker(ctx, cow, val.Size)
+	ls := cidlink.DefaultLinkSystem()
+	bsa := bsadapter.Adapter{Wrapped: s.bstore}
+	ls.SetReadStorage(&bsa)
+
+	content, err := car.NewCarV1StreamReader(
+		ctx,
+		&ls,
+		val.PayloadCid,
+		selectorparse.CommonSelector_ExploreAllRecursively,
+		car.WithDataPayloadSize(val.Size),
+	)
+	if err != nil {
+		return err
+	}
+	closer, _ := content.(io.Closer)
+
+	cancelContent := func(parentCtx context.Context) (err error) {
+		cancelComplete := make(chan struct{}, 1)
+		go func() {
+			// calling a Seek() on the SkipWriterReaderSeeker from go-car will cancel
+			// any existing writer and block until that cancel is complete
+			if closeErr := closer.Close(); closeErr != nil && !errors.Is(closeErr, context.Canceled) {
+				// context.Canceled can come up through the http Request from a socket close, ignore it
+				err = closeErr
+			}
+			cancelComplete <- struct{}{}
+		}()
+		select {
+		case <-parentCtx.Done():
+			return parentCtx.Err()
+		case <-cancelComplete:
+		}
+		return err
+	}
```

Comment:
nit: I'd suggest moving the cancelContent function inside of `sendCar` as it's not used anywhere else